EssentialsDiscord/src/main/java/net/essentialsx/discord/listeners/BukkitListener.java

```diff
@@ -124,21 +129,45 @@ public void onVanishStatusChange(VanishStatusChangeEvent event) {
             return;
         }
         if (event.getValue()) {
-            sendJoinQuitMessage(event.getAffected().getBase(), ChatColor.YELLOW + event.getAffected().getName() + " left the game", false);
+            sendJoinQuitMessage(event.getAffected().getBase(), ChatColor.YELLOW + event.getAffected().getName() + " left the game", "quit");
             return;
         }
-        sendJoinQuitMessage(event.getAffected().getBase(), ChatColor.YELLOW + event.getAffected().getName() + " joined the game", true);
+        sendJoinQuitMessage(event.getAffected().getBase(), ChatColor.YELLOW + event.getAffected().getName() + " joined the game", "join");
     }
 
-    public void sendJoinQuitMessage(final Player player, final String message, boolean join) {
-        sendDiscordMessage(join ? MessageType.DefaultTypes.JOIN : MessageType.DefaultTypes.LEAVE,
-                MessageUtil.formatMessage(join ? jda.getSettings().getJoinFormat(player) : jda.getSettings().getQuitFormat(player),
-                        MessageUtil.sanitizeDiscordMarkdown(player.getName()),
-                        MessageUtil.sanitizeDiscordMarkdown(player.getDisplayName()),
-                        MessageUtil.sanitizeDiscordMarkdown(message),
-                        jda.getPlugin().getEss().getOnlinePlayers().size() - (join ? 0 : 1),
-                        jda.getPlugin().getEss().getUsers().getUserCount()),
+    public void sendJoinQuitMessage(final Player player, final String message, String type) {
+        switch (type) {
```

Comment:
reduce the duplication here, just put the enum and player count in the switch case and leave the rest outside.

Reply:
Sorry for the delay, pushed this